hummingbot/connector/derivative/dydx_perpetual/dydx_perpetual_derivative.py

```diff
@@ -0,0 +1,1076 @@
+                return False
+        except Exception as e:
+            self.logger().warning(f"Failed to cancel order {client_order_id}")
+            self.logger().info(e)
+            return False
+
+    async def cancel_all(self, timeout_seconds: float) -> List[CancellationResult]:
+        cancellation_queue = self._in_flight_orders.copy()
+        if len(cancellation_queue) == 0:
+            return []
+
+        order_status = {o.client_order_id: o.is_done for o in cancellation_queue.values()}
+
+        def set_cancellation_status(oce: OrderCancelledEvent):
+            if oce.order_id in order_status:
+                order_status[oce.order_id] = True
+                return True
+            return False
+
+        cancel_verifier = LatchingEventResponder(set_cancellation_status, len(cancellation_queue))
+        self.add_listener(ORDER_CANCELLED_EVENT, cancel_verifier)
+
+        for order_id, in_flight in cancellation_queue.items():
+            try:
+                if order_status[order_id]:
+                    cancel_verifier.cancel_one()
+                elif not await self.cancel_order(order_id):
+                    # this order did not exist on the exchange
+                    cancel_verifier.cancel_one()
+                    order_status[order_id] = True
+            except Exception:
+                cancel_verifier.cancel_one()
+                order_status[order_id] = True
+
+        await cancel_verifier.wait_for_completion(timeout_seconds)
+        self.remove_listener(ORDER_CANCELLED_EVENT, cancel_verifier)
+
+        return [CancellationResult(order_id=order_id, success=success) for order_id, success in order_status.items()]
+
+    def get_fee(self,
+                base_currency: str,
+                quote_currency: str,
+                order_type: OrderType,
+                order_side: TradeType,
+                amount: Decimal,
+                price: Decimal):
+        is_maker = order_type is OrderType.LIMIT
+        return estimate_fee("dydx_perpetual", is_maker)
+
+    # ==========================================================
+    # Runtime
+    # ----------------------------------------------------------
+
+    def start(self, clock: Clock, timestamp: float):
+        super().start(clock, timestamp)
+
+    def stop(self, clock: Clock):
+        super().stop(clock)
+
+    async def start_network(self):
+        await self.stop_network()
+        self._order_book_tracker.start()
+        if self._trading_required:
+            self._polling_update_task = safe_ensure_future(self._polling_update())
+            self._user_stream_tracker_task = safe_ensure_future(self._user_stream_tracker.start())
+            self._user_stream_event_listener_task = safe_ensure_future(self._user_stream_event_listener())
+
+    def _stop_network(self):
+        self._order_book_tracker.stop()
+        self._polling_update_task = None
```

Comment:
```suggestion
        self._order_book_tracker.stop()
        if self._polling_update_task is not None:
            self._polling_update_task.cancel()
            self._polling_update_task = None
```

Reply:
Resolved in https://github.com/CoinAlpha/hummingbot/pull/3130/commits/4c213f3d8146a51567ee6af4afcc92c3195ca119